src/App.tsx

```diff
@@ -1,60 +1,139 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [title, setTitle] = useState('');
+  const [titleError, setTittleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const addTodo = (id: number) => {
+    const foundedUser = usersFromServer.find(user => user.id === id);
+    const uniqueId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+    if (title.trim().length === 0) {
```

Comment:
should work like this too
```suggestion
    if (title.trim().length) {
```

Reply:
no it doesn't 😢